deep_qa/data/datasets/entailment/snli_dataset.py

```diff
@@ -0,0 +1,35 @@
+from typing import List
+import json
+
+from overrides import overrides
+
+from ..dataset import TextDataset, count_labels
+from ...instances import TextInstance
+from ....common.params import Params
+
+
+class SnliDataset(TextDataset):
+
+    def __init__(self, instances: List[TextInstance], params: Params=None):
+        super(SnliDataset, self).__init__(instances, params)
+
+    @staticmethod
+    @overrides
+    def read_from_file(filename: str, instance_class, params: Params=None):
+
+        instances = []
+        for line in open(filename, 'r'):
+            example = json.loads(line)
+
+            # TODO(mark) why does this not match snli? Fix.
```

Comment:
Apart from this also requires redoing the data which we have in `/efs`, so i'm going to leave this out of this PR.